src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -441,16 +442,63 @@ namespace OpenLoco::World::MapGenerator
         call(0x004597FD, regs);
     }
 
+    template<typename Func>
+    static void generateMiscBuilding(const BuildingObject* buildingObj, const size_t id, Func&& predicate)
+    {
+        uint8_t randomComponent = getGameState().rng.randNext(0, buildingObj->var_9F / 2);
+        uint8_t staticComponent = buildingObj->var_9F - (buildingObj->var_9F / 4);
+
+        uint8_t amountToBuild = randomComponent + staticComponent;
+        if (amountToBuild == 0)
+            return;
+
+        for (auto i = 0U; i < amountToBuild; i++)
+        {
+            for (auto attemptsLeft = 200; attemptsLeft > 0; attemptsLeft--)
+            {
+                // NB: coordinate selection has been simplified compared to vanilla
+                auto randomX = getGameState().rng.randNext(2, 382);
```

Comment:
is there a way to get the map size limits here? this will need to be looked up for when maps are variable sizes

Reply:
Good point. Yes, map constants should be used.